These are defined on ppc* in included headers but their values refer to things
that are in headers which are not included. Additionally, ELF_GREG_TYPE refers
to a typedef, which is not a constant value. So undef them, just like some
others are already undef'd for other platforms above that.

--- a/gen_constants-inl.h
+++ b/gen_constants-inl.h
@@ -24,3 +24,8 @@
 // build errors on such broken systems.
 #undef BLKTRACESETUP
 #undef FS_IOC_FIEMAP
+#undef ELF_ARCH
+#undef ELF_CLASS
+#undef ELF_DATA
+#undef ELF_GREG_TYPE
+#undef FIOQSIZE
